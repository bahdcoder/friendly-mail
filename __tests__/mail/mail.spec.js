const Mail = require('../../src/index')

describe('The Mail class', () => {
  it('instantiates with preferred config if its passed as second argument ', () => {
    const config = require('../../mail.config')
    config.TEST_CONFIG = 'TEST_CONFIG'
  
    const mail = new Mail('confirm-account', config)

    expect(mail.Config.TEST_CONFIG).toBe('TEST_CONFIG')
  })

  it('instantiates with a custom config', () => {
    process.env.MAIL_CONFIG_FILE_PATH =
      '__tests__/__mocks__/config/custom-mail.config.js'

    const mail = new Mail()

    expect(mail.Config.testing).toBe(true)
    expect(mail._configFilePath).toEqual(
      '__tests__/__mocks__/config/custom-mail.config.js'
    )
  })

  it('sets driver instance once instantiated', () => {
    expect(new Mail()._driverInstance).toBeTruthy()
  })

  it('can set `inReplyTo` for message to be sent', () => {
    const mail = new Mail().inReplyTo('test@mail.ru')

    expect(mail.mailerMessage.inReplyTo).toEqual('test@mail.ru')
  })

  it('can set `subject` for message to be sent', () => {
    const mail = new Mail()

    mail.subject('Test Mail')
    expect(mail.mailerMessage.subject).toEqual('Test Mail')
  })

  it('can set `from` for message to be sent', () => {
    const mail = new Mail().inReplyTo('10122121112')

    expect(mail.mailerMessage.inReplyTo).toEqual('10122121112')
  })

  it('can build a complete mail message', () => {
    const mail = new Mail()

    mail
      .driverExtras()
      .data({
        name: 'Foo Bar',
        username: 'foo-bar-js'
      })
      .inReplyTo('10122121112')
      .embed('logo.png', 'logo')
      .from('foo@bar.com', 'Foo Bar')
      .to('jane@doe.com', 'Jane Doe')
      .cc('john@doe.com', 'John doe')
      .attachData('hello text', 'hello.txt')
      .bcc('admin@app.com', 'Administrator')
      .sender('mark@meyner.com', 'Mark Meyner')
      .replyTo('anne@meyner.com', 'Anne Meyner')
      .attach('absolute/path/to/file.jpg', { contentTpe: 'plain/text' })
      .alternative('**Hello**', { contentType: 'text/x-web-markdown' })

    expect(mail.mailerMessage).toMatchSnapshot()
  })
})
