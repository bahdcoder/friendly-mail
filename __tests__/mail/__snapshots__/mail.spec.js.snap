// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Mail class can build a complete mail message 1`] = `
Object {
  "alternatives": Array [
    Object {
      "content": "**Hello**",
      "contentType": "text/x-web-markdown",
    },
  ],
  "attachments": Array [
    Object {
      "cid": "logo",
      "path": "logo.png",
    },
    Object {
      "content": "hello text",
      "filename": "hello.txt",
    },
    Object {
      "contentTpe": "plain/text",
      "path": "absolute/path/to/file.jpg",
    },
  ],
  "bcc": Array [
    Object {
      "address": "admin@app.com",
      "name": "Administrator",
    },
  ],
  "cc": Array [
    Object {
      "address": "john@doe.com",
      "name": "John doe",
    },
  ],
  "extras": undefined,
  "from": Array [
    Object {
      "address": "foo@bar.com",
      "name": "Foo Bar",
    },
  ],
  "inReplyTo": "10122121112",
  "replyTo": Array [
    Object {
      "address": "anne@meyner.com",
      "name": "Anne Meyner",
    },
  ],
  "sender": Array [
    Object {
      "address": "mark@meyner.com",
      "name": "Mark Meyner",
    },
  ],
  "to": Array [
    Object {
      "address": "jane@doe.com",
      "name": "Jane Doe",
    },
  ],
}
`;
